addons/mail/models/mail_channel.py

```diff
@@ -282,6 +282,13 @@ def write(self, vals):
 
         if vals.get('group_ids'):
             self._subscribe_users()
+        # Reflect changes to every members when switching a channel to a mailing one or vice versa
+        if 'email_send' in vals:
+            notifications = []
+            for record in self:
+                for partner in record.channel_partner_ids:
+                    notifications.append([(self._cr.dbname, 'res.partner', partner.id), {'channel_id': record.id, 'type': 'toggle_mass_mailing'}])
+                self.env['bus.bus'].sendmany(notifications)
```

Comment:
This should be moved one indent level down, so that is is actually calling `sendmany` only once.